doc/config.rst

```diff
@@ -1617,6 +1617,20 @@ These options influence LaTeX output. See further :doc:`latex`.
 
    .. versionadded:: 1.4.5
 
+.. confval:: latex_use_latex_multicolumn
+
+   If ``False`` (default), the LaTeX writer uses for merged cells in grid
+   tables Sphinx's own macros. They have the advantage to allow the same
+   contents as in non-merged cells (inclusive of literal blocks, lists,
+   multiple paragraphs, ...). But they assume that the columns are separated
+   by the standard vertical rule. In case the :rst:dir:`tabularcolumns`
+   directive was employed to inject more macros (using LaTeX's mark-up of the
+   type ``>{..}``, ``<{..}``, ``@{..}``) the behaviour differs from LaTeX's
+   own ``\multicolumn`` macro. For example coloured columns keep same color
```

Comment:
I can try to think about how make Sphinx multi-column better compatible with this extra mark-up allowed via tabularcolumns, but this is currently hard. The problem is a complicated TeX/LaTeX one. My ``\sphinxstartmulticolumn`` really needs the column TeX templates inserted because that seems the only way to access the designed dimensions in ``p{..}`` columns. Else, one needs to hack quite deeper in LaTeX and tabulary/array/colortbl etc...

As a result currently it is assumed that the column separators are the standard ``|``  ones.